internal/config/validator.go

```diff
@@ -58,13 +58,69 @@ func (v *validator) Validate(resources ...client.ObjectList) error {
 			clusterResources.Nodes = append(clusterResources.Nodes, list.Items...)
 		}
 	}
+	clusterResources = resetfields(clusterResources)
 	_, err := For(clusterResources, v.validate)
 	if errors.As(err, &TransientError{}) { // we do not want to make assumption on ordering in webhooks.
 		return nil
 	}
 	return err
 }
 
+func resetfields(clusterResources ClusterResources) ClusterResources {
+	for i := range clusterResources.Peers {
+		clusterResources.Peers[i].Spec.BFDProfile = ""
+		clusterResources.Peers[i].Spec.PasswordSecret.Name = ""
+	}
+	for _, bgpAdv := range clusterResources.BGPAdvs {
+		for _, community := range bgpAdv.Spec.Communities {
+			if !ExistCommunity(community, clusterResources.Communities) {
+				bgpAdv.Spec.Communities = RemoveElement(bgpAdv.Spec.Communities, community)
```

Comment:
I am not sure about what happens if we remove an item / reassing while we iterate.
A better approach is to build the new communities slice while iterate, and replace bgpAdv.Spec.Communities with the new slice after the loop